### ISSUE TITLE
Add labels to pods select filters and input search by keyword

### ISSUE THREAD

**Issue by capybaraking** [pods-framework/pods#7561]
### Problem to Solve

Pods filters inputs should have pertinent labels to be WCAG compliant : https://www.w3.org/WAI/WCAG21/Understanding/labels-or-instructions.html

### Proposed Solution

Add labels with default "Filter by"+placeholder text. It would also be great to customize the text so we can adjust it in multilingual multisite or add context informations

### Screenshots / Screencast

Current view
<img width="972" height="109" alt="Image" src="https://github.com/user-attachments/assets/6f0721f9-9e9c-4a57-926c-69c6ce745e2c" />

Desired view

<img width="196" height="95" alt="Image" src="https://github.com/user-attachments/assets/f5ff9f95-6be0-487b-bce3-89163cf925f7" />
